test/TextProviderHOC.test.js

```diff
@@ -0,0 +1,28 @@
+import withTextProvider from '../src/TextProviderHOC';
+
+describe('<TextProviderHOC /> rendering', () => {
+  class SomeComponent extends React.Component {
+    constructor(props) {
+      super(props);
+    }
+
+    render() {
+      return <div>{this.props.id ? this.props.id : 'id'}</div>;
+    }
+  }
+
+  it('should render TextProviderHOC', () => {
+    const MockWithHOC = withTextProvider(SomeComponent);
```

Comment:
What the HOC `withTextProvider` does is to add the `getTextForKey` as a prop of the component. I don't see any checks for the same anywhere.

Using it like this doesn't make any sense to me. Can you explain the logic behind this ?
```jsx
<MockWithHOC id="id" />
```

Reply:
Added id to check if the Component will have that prop.
Added a test case to check if getTextForKey is added as prop.